pkg/util/azureclient/decorators.go

```diff
@@ -0,0 +1,38 @@
+package azureclient
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest"
+)
+
+func DecorateSenderWithLogging(sender autorest.Sender) autorest.Sender {
+	loggingDecorator := LoggingDecorator()
+	return autorest.DecorateSender(sender, loggingDecorator, autorest.DoCloseIfError())
+}
+
+// LoggingDecorator returns a function which is used to wrap and modify
+// the behaviour of an autorest.Sender.
+// Azure Clients will have the sender wrapped by that function
+// in order to intercept http calls using our custom RoundTripper (through the adapter).
```

Comment:
This comment (which I've likely misunderstood) seems to be explaining what the following code does and (weak challenge) seems unneeded. What do you think?

Reply:
In general terms I do not like to put comments that explain what the code does (but instead why-> explaining what code can not explain).

In that case is different. I decided to explain the what as the usage of this wrapper function is new and can be a bit misleading to people not used to the azure sender thingy. I just added it tho clarify where the function is intended to be used. I am happy to delete the comment if we find it trivial.